logger/logger.go

```diff
@@ -7,31 +7,101 @@ package logger
 import (
 	"fmt"
 	"os"
+	"strings"
 	"time"
 )
 
+var requestedLevel LogLevel
+
+// LogLevel type
+type LogLevel uint32
+
+const (
+	// FatalLevel should be used in fatal situations, the app will exit
+	FatalLevel LogLevel = iota
+
+	// ErrorLevel should be used when someone should really look at the error
+	ErrorLevel
+
+	// InfoLevel should be used during normal operations
+	InfoLevel
+
+	// DebugLevel should be used only during development
+	DebugLevel
+)
+
+// AllLevels exposes all logging levels
+var AllLevels = []LogLevel{
+	FatalLevel,
+	ErrorLevel,
+	InfoLevel,
+	DebugLevel,
+}
+
+// Convert the Level to a string.
+func (level LogLevel) String() string {
+	switch level {
+	case DebugLevel:
+		return "DEBUG"
+	case InfoLevel:
+		return "INFO"
+	case ErrorLevel:
+		return "ERROR"
+	case FatalLevel:
+		return "FATAL"
+	}
+
+	return "UNKNOWN"
+}
+
+// SetLevel can be used to set the log level
+func SetLevel(wantedLevel string) {
+	if strings.ToUpper(wantedLevel) == DebugLevel.String() {
+		requestedLevel = DebugLevel
+	} else if strings.ToUpper(wantedLevel) == InfoLevel.String() {
+		requestedLevel = InfoLevel
+	} else if strings.ToUpper(wantedLevel) == ErrorLevel.String() {
+		requestedLevel = ErrorLevel
+	} else if strings.ToUpper(wantedLevel) == FatalLevel.String() {
+		requestedLevel = FatalLevel
+	} else {
+		formatMessage(ErrorLevel, fmt.Sprintf("Unknown log level provided: %s, using default level", wantedLevel))
+		requestedLevel = DebugLevel
+	}
```

Comment:
What about using a switch/case statements here instead of all these `if`?

Reply:
I will refactor.